setup.sh

```diff
@@ -232,27 +232,60 @@ setup_networking() {
 		cidr=$(echo $TINKERBELL_NETWORK | grep -Eo "\/[[:digit:]]+" | grep -v "^$" | tr -d "/")
 		case "$1" in 
 			ubuntu)
-				if [ ! -f /etc/network/interfaces ] ; then
-					echo "$ERR file /etc/network/interfaces not found"
-					exit 1
-				fi
+				v16=16.04
+				version=$(. /etc/os-release && echo "$VERSION_ID")
+				if (( $(echo "$version > $v16" | bc -l) )); then
+					echo "$INFO found Ubuntu $version, using 'netplan' for network configuration"
+
+					# this is required to fix the bug descibed here
+					# https://bugs.launchpad.net/netplan/+bug/1810043
+					apt-get install -y --no-install-recommends netplan.io >> /dev/null 
+					cat >/etc/netplan/${TINKERBELL_NETWORK_INTERFACE}.yaml <<EOF
+---
+network:
+  version: 2
+  renderer: networkd
+  ethernets:
+    $TINKERBELL_NETWORK_INTERFACE:
+      addresses:
+        - $TINKERBELL_HOST_IP/$cidr
+        - $TINKERBELL_NGINX_IP/$cidr
+EOF
+					ip link set $TINKERBELL_NETWORK_INTERFACE nomaster
+					netplan apply
+					echo "$INFO waiting for the network configuration to be applied by systemd-networkd"
```

Comment:
I think there is probably a systemd service we could wait on with something like:

```
while ! systemctl is-active netplan-....; do
  sleep 1
done
```

or perhaps some other unit. This would ensure we don't move past this step if it failed.